doc/source/development/maintaining.rst

```diff
@@ -345,7 +345,126 @@ The benchmarks are scheduled by Airflow. It has a dashboard for viewing and debu
 Release process
 ---------------
 
-The process for releasing a new version of pandas can be found at https://github.com/pandas-dev/pandas-release
+The release process makes a snapshot of pandas (a git commit) available to users with
+a particular version number. After the release the new pandas version will be available
+in the next places:
+
```

Comment:
It would be nice to add a prerequisites section, detailing what sort of permissions you need to release and how to ask for them.